Celeste.Mod.mm/Mod/Core/CoreModule.cs

```diff
@@ -224,8 +224,30 @@ public class CoreModule : EverestModule {
                     if (Everest.Loader.Delayed.Count > 0) {
                         menu.Add(new TextMenuExt.SubHeaderExt(Dialog.Clean("modoptions_coremodule_notloaded_a")) { HeightExtra = 0f, TextColor = Color.OrangeRed });
                         menu.Add(new TextMenuExt.SubHeaderExt(Dialog.Clean("modoptions_coremodule_notloaded_b")) { HeightExtra = 0f, TextColor = Color.OrangeRed });
-                        foreach (Tuple<EverestModuleMetadata, Action> mod in Everest.Loader.Delayed)
-                            menu.Add(new TextMenuExt.SubHeaderExt(mod.Item1.Name + " | v." + mod.Item1.VersionString) { HeightExtra = 0f, TextColor = Color.PaleVioletRed });
+
+                        foreach (Tuple<EverestModuleMetadata, Action> mod in Everest.Loader.Delayed) {
+                            string missingDepsString = "";
+                            if(mod.Item1.Dependencies != null) {
+                                // check for missing dependencies
+                                List<EverestModuleMetadata> missingDependencies = mod.Item1.Dependencies
+                                    .FindAll(dep => !Everest.Loader.DependencyLoaded(dep));
+
+                                if(missingDependencies.Count != 0) {
+                                    // format their names and versions, and join all of them in a single string
+                                    missingDepsString = string.Join(", ", missingDependencies.Select(dependency => dependency.Name + " | v." + dependency.VersionString));
+
+                                    // ensure that string is not too long, or else it would break the display
+                                    if (missingDepsString.Length > 40) {
+                                        missingDepsString = missingDepsString.Substring(0, 40) + "...";
+                                    }
+
+                                    // wrap that in a " ({list} not found)" message
+                                    missingDepsString = $" ({missingDepsString} {Dialog.Clean("modoptions_coremodule_notloaded_notfound")})";
```

Comment:
Appending to a string is slow. You might want to use https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder instead, but again, not sure it matters if it only executes on startup.

Reply:
You're right that appending to a string is "slow", but just like the Linq usage, interpolated strings are much, *much* easier to maintain than the "faster" StringBuilder, and the overhead is once again negligible here.